presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergAbstractMetadata.java

```diff
@@ -904,8 +905,16 @@ public void renameColumn(ConnectorSession session, ConnectorTableHandle tableHan
         Table icebergTable = getIcebergTable(session, icebergTableHandle.getSchemaTableName());
         Transaction transaction = icebergTable.newTransaction();
         transaction.updateSchema().renameColumn(columnHandle.getName(), target).commit();
-        if (icebergTable.spec().fields().stream().map(PartitionField::sourceId).anyMatch(sourceId -> sourceId == columnHandle.getId())) {
-            transaction.updateSpec().renameField(columnHandle.getName(), target).commit();
+        List<PartitionField> matchingFields = icebergTable.spec().fields().stream()
+                .filter(field -> field.sourceId() == columnHandle.getId())
+                .collect(toList());
+        if (!matchingFields.isEmpty()) {
+            for (PartitionField field : matchingFields) {
+                String transform = field.transform().toString();
+                transaction.updateSpec().renameField(
+                        getPartitionColumnName(columnHandle.getName(), transform),
+                        getPartitionColumnName(target, transform)).commit();
+            }
```

Comment:
```suggestion
        icebergTable.spec().fields().stream()
                .filter(field -> field.sourceId() == columnHandle.getId())
                .forEach(field -> {
                    String transform = field.transform().toString();
                    transaction.updateSpec().renameField(field.name(), getPartitionColumnName(target, transform)).commit();
                });
```

Fetching the original name directly from the partition field seems more robust. Besides, we can write this logic entirely using Stream-style code.

Reply:
Thanks.